### PR TITLE
Refactor the Compiler.process function

### PR DESCRIPTION
**Pull request by tomekb234**
This makes `Compiler.process` generally more readable. The interfaces for classes directly used by this function have been slightly changed, mostly by moving `diagnostics` parameter from the used function to its class' constructor. One notable change is that the vague class name `ControlFlow` has been changed to `ControlFlowPlanner` (because that's what this class basically does - it arranges the control flow of the program given its AST).

### CODE REVIEW

File: app/src/main/kotlin/compiler/Compiler.kt [hunk 1]
```diff
@@ -39,81 +38,101 @@ class Compiler(val diagnostics: Diagnostics) {
...
 
             val (functionDetailsGenerators, generatorDetailsGenerators) = FunctionDependenciesAnalyzer.createCallablesDetailsGenerators(
-                astWithBuiltinFunctions,
+                program,
                 programProperties.variableProperties,
                 programProperties.functionReturnedValueVariables,
-                diagnostics.hasAnyError()
-            )
+                diagnostics.hasAnyErrors()
+            ) // NOTE: it probably shouldn't be FunctionDependenciesAnalyzer responsible for this
```
Comment (ignacy123) [inline app/src/main/kotlin/compiler/Compiler.kt:71]:
```suggestion
            ) // TODO: it probably shouldn't be FunctionDependenciesAnalyzer responsible for this
```

File: app/src/test/kotlin/compiler/intermediate/FunctionControlFlowPlannerTest.kt [hunk 2]
```diff
@@ -113,21 +113,22 @@ class FunctionControlFlowTest {
...
     }
 
-    private fun test(program: Program) = ControlFlow.createGraphForEachFunction(
+    private val testDiagnostics = object : Diagnostics {
+        override fun report(diagnostic: Diagnostic) {
+            diagnostics.add(diagnostic)
+        }
+
+        override fun hasAnyErrors(): Boolean {
+            throw RuntimeException("This method shouldn't be called")
```
Comment (ignacy123) [inline app/src/test/kotlin/compiler/intermediate/FunctionControlFlowPlannerTest.kt:122]: Maybe NotImplementedError instead?

File: test_all.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-./gradlew test "$@"
+./gradlew "$@" test install
```
Comment (ignacy123) [inline test_all.sh:2]: Please update CI config to reflect this

**Review by ignacy123 — CHANGES_REQUESTED** [review]

File: app/src/test/kotlin/compiler/intermediate/FunctionControlFlowPlannerTest.kt (hunk 2 above)
Comment (tomekb234) [reply app/src/test/kotlin/compiler/intermediate/FunctionControlFlowPlannerTest.kt]: I'm not the author of this code, but sure.

File: test_all.sh (hunk 3 above)
Comment (tomekb234) [reply test_all.sh:2]: Done.

**Review by ignacy123 — APPROVED** [review]

**Pull request merged**